Dump button, dumps the grid to linked inventory

ChromaPIE opened the pull request:
<img width="354" height="180" alt="图片" src="https://github.com/user-attachments/assets/17edce15-7fcf-48a0-b14d-d3f515397da3" />

A button on Crafting Station's UI, sweeps stuff in the grid into the adjacent inventory

### WARNING
100% vibe coded & verified. See it as an issue if the impletation is shit.

ChromaPIE committed:
dump button

Ruling-0 committed:
Merge branch 'master' into dumpbutton

Ruling-0 reviewed (commented):
Sorry for the long review delay! Just a couple code quality things I think should be addressed.

Ranzuu commented:
Please make this a texture, not hardcoded graphic. At least something like this, 10x10 to match the bogo sorter icons.
| Button |
| - |
| <img width="10" height="10" alt="10" src="https://github.com/user-attachments/assets/0a4a540a-b096-4a09-a334-277a3efb6b57" />|

Dream-Master committed:
Merge branch 'master' into dumpbutton

ChromaPIE committed:
improve things

ChromaPIE commented:
done and tested. again the change are also vibe coded and thus needs verifying

Dream-Master committed:
Merge branch 'master' into dumpbutton

Dream-Master committed:
Merge branch 'master' into dumpbutton

Ruling-0 reviewed (changes requested):
You may use what tools you want, but please test things in game to ensure they work. Right now, the button renders as missing texture and, even fixing that, it looks different from other base-game buttons.

Dream-Master committed:
Merge branch 'master' into dumpbutton

ChromaPIE committed:
resolve

UltraProdigy committed:
optimize image

Dream-Master committed:
Merge branch 'master' into dumpbutton

Ruling-0 reviewed (commented):
The button highlight should function like this:

<img width="543" height="402" alt="Image" src="https://github.com/user-attachments/assets/b92f0460-d598-46a0-91f2-fd6ba5f50a9c" />

Dream-Master committed:
Merge branch 'master' into dumpbutton

Dream-Master committed:
Merge branch 'master' into dumpbutton

Dream-Master committed:
Merge branch 'master' into dumpbutton

Ranzuu committed:
Refresh the Dump Button Funcionality
- removed the standalone class and moved it into CraftingStationGui
- removed the standalone asse icon and moved it into icons.png (normal+hover state)
- added button tooltip

Ranzuu commented:
<img width="782" height="197" alt="image" src="https://github.com/user-attachments/assets/9fdfac69-00c2-4743-b4f1-963b2720e8a1" />

<img width="110" height="63" alt="image" src="https://github.com/user-attachments/assets/47acacd0-d10e-4346-9b74-70f9054a1f90" />

Ranzuu committed:
Merge branch 'master' into dumpbutton

Ranzuu committed:
dump button positioning
- 5px from each border

Ranzuu commented:
Imo this looks better
<img width="112" height="103" alt="image" src="https://github.com/user-attachments/assets/4c234715-d409-4ef9-b1b9-006a4d513ff0" />

UltraProdigy committed:
optimize image

UltraProdigy commented:
Image is good now 👍

Ruling-0 commented:
> Imo this looks better <img alt="image" width="112" height="103" src="https://private-user-images.githubusercontent.com/66495944/583855681-4c234715-d409-4ef9-b1b9-006a4d513ff0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzczNjM0MTQsIm5iZiI6MTc3NzM2MzExNCwicGF0aCI6Ii82NjQ5NTk0NC81ODM4NTU2ODEtNGMyMzQ3MTUtZDQwOS00ZWY5LWIxYjktMDA2YTRkNTEzZmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDI4VDA3NTgzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5YjUzNmFjMjU5N2ViNDRlNTE3YzgxY2QwZjkzYjBlYTEyZjc5Y2IyNWY2OWNmOWU4ODQxYzE1MWI4OWQyYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.BnD9z7QRjV39CNzy9bcXehXkuapvbIuRrG4gYbMOdHg">

Thank you for picking this up -- it's a good feature! The main crux of the issue I have is we shouldn't have 3 different standards for buttons *in the same UI*.

This is the base UI, not hovering any buttons (in daily). Looks fine:
<img width="507" height="238" alt="image" src="https://github.com/user-attachments/assets/641ac3c4-9d80-4319-9b51-08d39ccc75aa" />

This is hovering each of the three types:
<img width="280" height="98" alt="image" src="https://github.com/user-attachments/assets/63fe2eca-30df-4633-9383-1fd506c0d6f5" />
<img width="191" height="70" alt="image" src="https://github.com/user-attachments/assets/8e08e92e-7919-4087-a302-958a5e136431" />
<img width="110" height="47" alt="image" src="https://github.com/user-attachments/assets/01e527ec-a227-475a-8eee-f563e2ef01a6" />

This button looks closer to the bogosort buttons, so using their highlight scheme would be sufficient.

Ranzuu committed:
Dump Button Styling

Ranzuu committed:
Merge branch 'dumpbutton' of https://github.com/chromapie/tinkersconstruct into pr/ChromaPIE/228

Ranzuu commented:
✨ 

<img width="196" height="313" alt="image" src="https://github.com/user-attachments/assets/751690eb-5e2e-400f-a7c3-0715bea9c80b" />

Ranzuu committed:
Optimize Images

UltraProdigy committed:
optimize image
